lib/java/com/google/android/material/slider/BaseSlider.java

```diff
@@ -1699,10 +1699,6 @@ public boolean onTouchEvent(@NonNull MotionEvent event) {
         break;
       case MotionEvent.ACTION_MOVE:
         if (!thumbIsPressed) {
-          // Check if we're trying to scroll instead of dragging this Slider
-          if (isInScrollingContainer() && abs(x - touchDownX) < scaledTouchSlop) {
-            return false;
-          }
```

Comment:
@Subuday I don't think we can just remove this. It was added to fix issues with vertical scrolling:

https://github.com/material-components/material-components-android/issues/832
https://github.com/material-components/material-components-android/pull/833

Maybe instead of `isInScrollingContainer()` we could use `View#canScrollVertically` and `View#canScrollHorizontally` on the scrolling parent somehow for this x touch check.

cc: @cketcham

Reply:
I'm wondering if it's even a good idea to put a slider in a horizontally scrolling view? It wouldn't be possible to distinguish between an attempt to change the value on the slider or scroll. This would force the user to touch outside the Slider to perform a scroll on the parent.

Reply:
That's a good point, but I think the same thing could be said of putting a Slider in a TabLayout/ViewPager. I feel like it should be possible, but to swipe or scroll the user just has to touch outside the Slider like you said.

And maybe a small optimization could be that if the Slider is at its full left/right position and can't move its thumb any more depending on the direction of the swipe/scroll, then that would result in a swipe/scroll.

Reply:
@dsn5ft Yes, it's a good idea to check if it's in vertical or horizontal scrolled view.

Reply:
@cketcham As for my case, it's dragging card like in Badoo which implemented using RecyclerView has slider inside.